File: code/dobot_quad_sdk-main/high_level/cpp/e2_get_current_state.cpp
// Example 2: Get the current FSM state name.
// Usage: ./e2_get_current_state [server_address]

#include "robot_client.h"

int main(int argc, char** argv)
{
    robot::Client client(argc > 1 ? argv[1] : "192.168.5.2:50051");
    auto state = client.get_current_state_name();

    if (state.empty()) {
        std::cerr << "Failed to get state" << std::endl;
        return 1;
    }

    std::cout << "Current state: " << state << std::endl;
    return 0;
}

File: code/dobot_quad_sdk-main/high_level/python/examples/e2_get_current_state.py
#!/usr/bin/env python3
"""Example 2: Get the current FSM state name."""
import sys
from dobot_quad import RobotClient


def main():
    robot = RobotClient(sys.argv[1] if len(sys.argv) > 1 else "192.168.5.2:50051")
    state = robot.get_current_state_name()

    if state:
        print(f"Current state: {state}")
        return 0
    else:
        print("Failed to get state")
        return 1


if __name__ == "__main__":
    sys.exit(main())
